packages/@lwc/engine-dom/src/custom-elements/create-custom-element.ts

```diff
@@ -11,6 +11,16 @@ const cachedConstructors = new Map<string, CustomElementConstructor>();
 const elementsUpgradedOutsideLWC = new WeakSet<HTMLElement>();
 let elementBeingUpgradedByLWC = false;
 
+let formAssociatedCallbackToUse: LifecycleCallback | undefined;
+let formDisabledCallbackToUse: LifecycleCallback | undefined;
+let formResetCallbackToUse: LifecycleCallback | undefined;
+let formStateRestoreCallbackToUse: LifecycleCallback | undefined;
+
+const instancesToFormAssociatedCallbacks = new WeakMap<HTMLElement, LifecycleCallback>();
+const instancesToFormDisabledCallbacks = new WeakMap<HTMLElement, LifecycleCallback>();
+const instancesToFormResetCallbacks = new WeakMap<HTMLElement, LifecycleCallback>();
+const instancesToFormStateRestoreCallbacks = new WeakMap<HTMLElement, LifecycleCallback>();
+
```

Comment:
Updated this to match the [new way](https://github.com/salesforce/lwc/pull/3662/files#diff-aac76e0c764682bec843d12e45df947e550cec67c071b719a5fe0f3fd6535a0e) of including native `connectedCallback` / `disconnectedCallback`.